### PR TITLE
Manage Vault Secrets

### PR DESCRIPTION
**Pull request by adamgoose**
Enables users to manage vault secrets in git, and apply them across multiple Vault servers in Truss.

TODO:

- [x] Refactor Port Forwarding to enable multiple Vault commands to be sent to a given Port Forwarding session
- [x] Determine whether the tmp file was actually saved during `truss secrets edit`, and only write to Vault accordingly
- [x] Interactively prompt the user to push after an edit, and/or provide a flag/envvar
- [x] Add pull command
- [ ] Add tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,6 +47,48 @@ brew install instructure-bridge/truss-cli/truss-cli
...
+The `truss secrets` command makes it easier to manage secrets in git, and
+synchronize them across multiple Truss Vault servers. Start by creating a
+`secrets.yaml` file.
+
+```yaml
+# secrets.yaml
+transit-key-name: my-project
+environments:
+  edge-cmh: # As declared in ~/.truss.yaml
+    secret: ./secrets/edge-cmh # Relative to `pwd`
```
Comment (jgramoll) [inline README.md:63]: I would expect this to be the vault secret path, which would have nothing to do with pwd

File: truss/secrets_manager.go [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+		directory = home + "/.kube/"
+	}
+
+	environments := viper.GetStringMapString("environments")
+
+	return Kubectl(path.Join(directory, environments[environment])), nil
+}
+
+// Vault creates a proxied Vault client
+func (m SecretsManager) Vault(environment string) (*VaultCmd, error) {
```
Comment (jgramoll) [inline truss/secrets_manager.go:174]: should this be a private function?

File: README.md (hunk 1 above)
Comment (adamgoose) [reply README.md:63]: Yeah, the naming between "secret" and "path" is weird... One is supposed to be where the local file is stored, the other is where they're stored in Vault. Any suggestions for naming here?

File: truss/secrets_manager.go (hunk 2 above)
Comment (adamgoose) [reply truss/secrets_manager.go:174]: I thought about it, but I think not, since some of the other functions require a vault instance as an argument.

**Pull request merged**